tools/toolchains/iar.py

```diff
@@ -45,12 +46,18 @@ def __init__(self, target, notify=None, macros=None, build_profile=None,
                  build_dir=None):
         mbedToolchain.__init__(self, target, notify, macros, build_dir=build_dir,
                                build_profile=build_profile)
-        if target.core == "Cortex-M7F" or target.core == "Cortex-M7FD":
-            cpuchoice = "Cortex-M7"
-        elif target.core.startswith("Cortex-M23"):
-            cpuchoice = "8-M.baseline"
+        if target.core == "Cortex-M7FD":
+            cpuchoice = "Cortex-M7.fp.dp"
+        elif target.core == "Cortex-M7F":
+            cpuchoice = "Cortex-M7.fp.sp"
+        elif target.core.startswith("Cortex-M33FD"):
+            cpuchoice = "Cortex-M33.fp"
```

Comment:
Wonder why this is not fp.dp (double-prec) as in M7? Is this correct?

Reply:
D in case of Cortex-M33 is DSP and not double floating point.

Reply:
Which is a very bad idea, as I believe I've previously commented. Why are we making up non-standard inconsistent naming?

If you really want to have a single-letter name, "M33FE" would make more sense, as "E" is the letter historically used by ARM to represent the DSP Extensions.

But I would rather go with what all the tools are consistently doing - using the "no_fp" and "no_dsp" suffixes. We're in the odd situation where all 3 toolchains agree on that, and Mbed OS is being different.

Reply:
While performing some changes recently, I see why Mbed OS used its own letters, as we do not have consistency among all toolchaina and even between assembler and compilers. Flags are +nodsp / .no_dsp / .dsp / +dsp

Reply:
> single-letter name, "M33FE"

Yes can change this now to some other letter for DSP, as no target is added.  
@mbed-os-tools - thoughts on this please

Reply:
Please make the core names consistent with the ARM historical single letter abbreviations; use `E` for DSP.

Reply:
@theotherjimmy @kjbracey-arm - One more Query, for DSP and FPU.  Should it be `F-E`or `FE`
Floating point - F and FD looks fine, but with addition of E, `FDE` will be confusing. And will be allow any order here? like `FD-E` or `E-FD`?